queries/go.scm

```diff
@@ -222,3 +222,59 @@
     .
   )
 ) @anonymousFunction @namedFunction
+
+;; switch-based branch
+
+(
+  [
+    (default_case)
+    (expression_case)
+    (type_case)
+  ] @branch
+  (#trim-end! @branch)
+  (#insertion-delimiter! @branch "\n")
+)
+
+[
+  (type_switch_statement)
+  (expression_switch_statement)
+] @branch.iteration
+
+;; if-else-based branch
+
+;; first if in an if-else chain
+(
+  (
+    (if_statement
+      consequence: (block) @branch.end.endOf
+    ) @_if
+    (#not-parent-type? @_if if_statement)
+  )
+ @branch.start.startOf
+  (#insertion-delimiter! @branch.start.startOf " ")
+)
+
+;; internal if in an if-else chain
+(if_statement
+  "else" @branch.start.startOf
+  alternative: (if_statement
+    consequence: (block) @branch.end.endOf
+  )
+  (#insertion-delimiter! @branch.start.startOf " ")
+)
+
+;; final else branch in an if-else chain
+(
+  (if_statement
+    "else" @branch.start.startOf
+    alternative: (block)
+  ) @branch.end.endOf
+  (#insertion-delimiter! @branch.start.startOf " ")
+)
+
+;; iteration scope is always the outermost if statement
+(
+  (if_statement) @_if
+  (#not-parent-type? @_if if_statement)
+)
+ @branch.iteration
```

Comment:
as above, autoformatter is doing strange stuff when you put a capture on a group, but I'm not sure what that does for you here either

Reply:
done